src/mono/mono/component/debugger-agent.c

```diff
@@ -7424,7 +7424,111 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
 
 	return ERR_NONE;
 }
+static void
+set_jmc_breakpoint_on_methods (MonoClass *klass, GPtrArray *method_array, Modifier *modifier, EventRequest *req)
+{
+	MonoMethod *m;
+	mono_class_setup_methods (klass);
+	gpointer iter = NULL;
+	while ((m = mono_class_get_methods (klass, &iter))) {
+		bool found = FALSE;
+		guint n = 0;
+		while (n < method_array->len) {
+			if (m == (MonoMethod *)g_ptr_array_index (method_array, n))	{
+				found = TRUE;
+				break;
+			}
+		}
+		if (!found)	{
+			MonoBreakpoint* bp = mono_de_set_breakpoint (m, 0, req, NULL);
+			modifier->data.jmc->bps = g_slist_append (modifier->data.jmc->bps, bp);
+		}
+	} 
+}
 
+static ErrorCode
+jmc_modifier (int *step_thread_id, StepSize *size, StepDepth *depth, Modifier *modifier, EventRequest *req, guint8 *p, guint8 *end, Buffer *buf)
+{
+	ErrorCode err;
+	ERROR_DECL (error);
+	MonoDomain *domain = NULL;
+	MonoThread *step_thread;
+	*step_thread_id = decode_id (p, &p, end);
+	*size = (StepSize)decode_int (p, &p, end);
+	*depth = (StepDepth)decode_int (p, &p, end);
+	int module_len = decode_int (p, &p, end);
+	MonoMethod *paused_method = NULL;
+	modifier->data.jmc = (JMC_Modifier *)g_malloc0 (sizeof (JMC_Modifier));
+	if (*depth == STEP_DEPTH_OVER) {
+		g_assert (*step_thread_id);
+		err = get_object (*step_thread_id, (MonoObject**)&step_thread);
+		if (err != ERR_NONE) {
+			g_free (req);
```

Comment:
curious - why is this freed in this code path, but not for the others in this function?